src/prompts/search.eval.spec.ts

```diff
@@ -0,0 +1,129 @@
+      },
+      {
+        key: "metrics.latency",
+        value: "P95 latency increased from 200ms to 2500ms at 14:30 UTC",
+      },
+      {
+        key: "database.metrics",
+        value:
+          "MongoDB read operations showing increased I/O wait times, averaging 1500ms per query",
+      },
+      {
+        key: "system.logs",
+        value:
+          "High memory usage on database cluster primary node (92% utilized)",
+      },
+      {
+        key: "deployment.history",
+        value:
+          "Latest deployment at 14:15 UTC: Updated user authentication caching layer",
+      },
+      {
+        key: "marketing.events",
+        value: "New social media campaign launched at 13:00 UTC",
+      },
+      {
+        key: "system.updates",
+        value: "Routine security patches scheduled for next Tuesday",
+      },
+      {
+        key: "network.status",
+        value: "CDN edge node in Sydney reports optimal performance",
+      },
+      {
+        key: "monitoring.alerts",
+        value: "SSL certificate for dev.example.com expires in 25 days",
+      },
+      {
+        key: "system.disk",
+        value: "Backup disk cleanup completed successfully, 234GB freed",
+      },
+    ];
+
+    const question =
+      "What's causing the high latency in the /api/users endpoint?";
+
+    const [prompt, config] = searchContextPrompt(question, contextRows);
+
+    const { object: relevantContext } = await generateObject({
+      output: "array",
+      schema: z.object({
+        relevance: z.number(),
+        context: z.string(),
+      }),
+      ...config,
+      prompt,
+    });
+
+    const expected = [
+      {
+        relevance: 1,
+        context:
+          "MongoDB read operations showing increased I/O wait times, averaging 1500ms per query",
+      },
+      {
+        relevance: 1,
+        context: "Memory usage on database cluster primary node (92% utilized)",
+      },
+      {
+        relevance: 1,
+        context:
+          "Deployment of the user authentication caching layer at 14:15 UTC",
+      },
+    ];
+
+    const scores = await Promise.all([
+      Possible({
+        input: question,
+        output: JSON.stringify(relevantContext),
+        expected: JSON.stringify(expected),
+      }),
+      Factuality({
+        input: JSON.stringify(contextRows),
+        output: JSON.stringify(relevantContext),
+        expected: JSON.stringify(expected),
+      }),
+      AnswerRelevancy({
+        input: question,
+        output: JSON.stringify(relevantContext),
+        context: JSON.stringify(contextRows),
+      }),
+      Battle({
+        instructions: prompt,
+        output: JSON.stringify(relevantContext),
+        expected: JSON.stringify(expected),
+      }),
+    ]);
+
+    expect(scores[0].score).toBe(1);
+    expect(scores[1].score).toBeGreaterThanOrEqual(0.6);
+    expect(scores[2].score).toBeGreaterThanOrEqual(0.6);
+    expect(scores[3].score).toBeGreaterThanOrEqual(0.6);
```

Comment:
Will we get some extra details in the report if this fails?

Reply:
yes the matcher prints input, output, expected, rationale, matcher type and score